DnDCompanion/Services/Networking/NetworkClient.swift

```diff
@@ -0,0 +1,28 @@
+//
+//  NetworkClient.swift
+//  DnDCompanion
+//
+//  Created by Nick on 3/19/25.
+//
+
+import Foundation
+
+protocol NetworkClient {
```

Comment:
Protocols should match proper naming conventions.  I would suggest using the following categorization and naming guidelines for the protocols in Swift:

1. If the implementer is something, name the protocol with a noun, e.g. Sequence, View, Repository
2. If the implementer is doing something, name the protocol with an adjective ending with ing, e.g. Loading, Generating, Coordinating
3. If something is done to the implementer, name the protocol with an adjective ending with able or ible, e.g. Comparable, Codable, Cachable